app/controllers/prior_authority/steps/primary_quote_controller.rb

```diff
@@ -1,32 +1,39 @@
 module PriorAuthority
   module Steps
     class PrimaryQuoteController < BaseController
+      before_action :set_service_type
+
       def edit
         @form_object = PrimaryQuoteForm.build(
-          primary_quote,
+          record,
           application: current_application
         )
       end
 
       def update
-        record = primary_quote
         update_and_advance(PrimaryQuoteForm, as:, after_commit_redirect_path:, record:)
       end
 
       private
 
-      def primary_quote
-        record = current_application.primary_quote || current_application.build_primary_quote
-        record.service_type = record.custom_service_name if record.service_type == 'custom'
-        record
+      def record
+        current_application.primary_quote || current_application.build_primary_quote
       end
 
       def as
         :primary_quote
       end
 
       def additional_permitted_params
-        [:service_type_suggestion]
+        %i[service_type_suggestion service_type custom_service_name]
+      end
+
+      def set_service_type
+        return unless current_application.service_type == 'custom'
```

Comment:
any reason we are doing this here and not in the form object :)

Reply:
Laziness - I did the smallest refactor possible to get the tests passing again. I'll try moving it on Monday.